quipucords/api/insights_report/tests_insights_report.py

```diff
@@ -109,63 +110,70 @@ def setUp(self):
             id=1, report_id=1, report_version=self.report_version,
             status=DeploymentsReport.STATUS_COMPLETE,
             cached_insights=None,
-            cached_fingerprints=json.dumps(self.fingerprints))
+            cached_fingerprints=json.dumps(self.fingerprints),
+        )
 
+    @pytest.mark.slow
     def test_get_insights_report_200_exists(self):
         """Retrieve insights report."""
-        url = '/api/v1/reports/1/insights/'
-        expected_hosts = [{'unique_id': i} for i in range(10001)]
-        self.deployments_report.cached_insights = json.dumps(
-            expected_hosts)
-        self.deployments_report.save()
-        with patch('api.insights_report.view.get_object_or_404',
-                   return_value=self.deployments_report):
-            response = self.client.get(url)
-        self.assertEqual(response.status_code, 200)
+        deployment_report = DeploymentReportFactory(
+            number_of_fingerprints=10_001,
+            status=DeploymentsReport.STATUS_COMPLETE,
+        )
```

Comment:
Running with 10k +1 fingerprints is definitely an overkill. on GH actions this is taking 11min lol

because of the serializer we need to properly create objects in the database - instead of simple dicts = `{"unique_id": <some number>}`

I didn't mock the setting controlling the number of hosts to keep the spirit of the original implementation, but this approach is too slow.

I think I'll change that. What do you think @gtanzillo @nicolearagao ?

Reply:
@bruno-fs I think you should definitely mock the number of hosts to reduce the time. 11 min is way too long for a test running on GH.